### PR TITLE
Python style chained comparison operators.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib-clay/operators/operators.clay [hunk 1]
```diff
@@ -247,6 +248,10 @@ inline overload greater?(a, b) = lesser?(b, a);
 
 inline overload greaterEquals?(a, b) = not lesser?(a, b);
 
+inline overload compare?(expr1, fn, expr2, ..args) : Bool
+    = if(not compare?(expr1, fn, expr2)) false else compare?(expr2, ..args);
```
Comment (jckarter) [inline lib-clay/operators/operators.clay]: This could be simplified a bit to `fn(expr1, expr2) and compare?(expr2, ..args)`.

Comment (ghost) [inline lib-clay/operators/operators.clay]: That's how I originally had it,  but then all comparisons in the chain 
are evaluated.

Comment (jckarter) [inline lib-clay/operators/operators.clay]: Are you sure about that? `and` short circuits.

Comment (ghost) [inline lib-clay/operators/operators.clay]: Hmmm, forgot about that. I'll change it back.

**Pull request merged**